searchcore/src/vespa/searchcore/proton/server/searchview.h

```diff
@@ -8,7 +8,7 @@
 
 namespace proton {
 
-class SearchView : public ISearchHandler
+class SearchView : public ISearchHandler, public std::enable_shared_from_this<ISearchHandler>
```

Comment:
A class inheriting from enable_shared_from_this should pass itself as template parameter.

Also consider making the constructors private and replace them with static create functions returning shared pointers to ensure that objects are actually owned by shared pointers.

Reply:
Fixed.